src/views/docs/en/guides/developer-experience/sharing-code.md

```diff
@@ -9,25 +9,82 @@ sections:
   - src/views
 ---
 
-Architect makes it easy to share code across many Lambda functions. Apps most commonly need to share business logic and view templates so Architect provides `@shared` and `@views` capability. Architect copies the contents of `src/shared` into all Lambdas and `src/views` into Lambda functions wired to respond to `@http` GET requests.
+Architect makes it easy to share code across Lambda functions. The most common use case is to share business logic and view templates. Architect has [`@shared`](../../reference/project-manifest/shared) and [`@views`](../../reference/project-manifest/views) pragmas to provide this functionality.
 
-Example `app.arc`
+Architect automatically copies the contents of `src/shared` into all Lambdas and `src/views` into `@http` GET Lambda functions.
+
+## Example usage
+
+Given a simple `app.arc` where `@shared` and `@views` are declared:
 
 ```arc
 @app
 myapp
 
 @shared
-src src/shared # this is the default
 
 @views
-src src/views # this is the default
 
 @http
 get /
 post /like
 ```
 
+Where utility code lives in `./src/shared` and common view code in `.src/views`:
+
+```sh
+.
+├── src
+│   ├── http
+│   │   ├── get-index/index.js
+│   │   └── post-like/index.js
+│   └── shared
+│       └── authenticate.js
+│   └── views
+│       └── document.js
+└── app.arc
+```
+
+`get-index` can use shared code by requiring it from `@architect/shared/<file>` and views code from `@architect/views/<file>`:
+
+```js
+// get-index/index.js
+const auth = require('@architect/shared/authenticate')
+const document = require('@architect/views/document')
+```
+
+The `post-like` route has access to shared code as well, but not views because it does not respond to a GET request.
+
+```js
+// post-like/index.js
+const auth = require('@architect/shared/authenticate')
+```
+
+## Custom shared paths
+
+The default shared and views directories can be overridden:
+
+```arc
+@app
+myapp
+
+@shared
+src path/to/shared
+
+@views
+src path/to/views
+```
+
+They are still required in the same way:
+
+```js
+// get-index/index.js
+const auth = require('@architect/shared/authenticate')
+const document = require('@architect/views/document')
+```
```

Comment:
This section is very JS specific. Should we also add sections or tabbed code blocks that show folks how to do this in Ruby and Python?

Reply:
Good idea! I've been testing Ruby shared code this AM and trying to iron out load paths still. Python next.